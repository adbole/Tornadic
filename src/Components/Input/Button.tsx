const Button = ({ 
    children, 
    onClick,
    ...excess 
}: React.ButtonHTMLAttributes<HTMLButtonElement> ) => (
    <button type="button" className="primary" onClick={onClick} {...excess}>{children}</button>
);

export default Button;