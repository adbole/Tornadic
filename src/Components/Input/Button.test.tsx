import { render } from "@testing-library/react";

import Button from "./Button";


test("Matches default primary snapshot", () => {
    const { container } = render(<Button>Test</Button>);

    expect(container).toMatchSnapshot();
});

test("Matches the secondary snapshot", () => {
    const { container } = render(<Button varient="secondary">Test</Button>);

    expect(container).toMatchSnapshot();
});


test("Matches the transparent snapshot", () => {
    const { container } = render(<Button varient="transparent">Test</Button>);

    expect(container).toMatchSnapshot();
});
