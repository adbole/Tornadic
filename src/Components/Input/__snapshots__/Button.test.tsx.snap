// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches default primary snapshot 1`] = `
.emotion-0 {
  -webkit-transition: filter 0.3s ease;
  transition: filter 0.3s ease;
  border: none;
  border-radius: var(--input-border-radius);
  font-size: inherit;
  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.25);
  margin: 5px 0px;
  color: inherit;
  background-color: var(--primary);
  padding: 5px 15px;
}

.emotion-0:hover,
.emotion-0:focus {
  cursor: pointer;
}

.emotion-0:active {
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
}

.emotion-0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(0.5);
  filter: grayscale(0.5);
}

<div>
  <button
    class="emotion-0 emotion-1"
    type="button"
  >
    Test
  </button>
</div>
`;

exports[`Matches the secondary snapshot 1`] = `
.emotion-0 {
  -webkit-transition: filter 0.3s ease;
  transition: filter 0.3s ease;
  border: none;
  border-radius: var(--input-border-radius);
  font-size: inherit;
  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.25);
  margin: 5px 0px;
  color: inherit;
  background-color: var(--background-layer);
  padding: 5px 15px;
}

.emotion-0:hover,
.emotion-0:focus {
  cursor: pointer;
}

.emotion-0:active {
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
}

.emotion-0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(0.5);
  filter: grayscale(0.5);
}

<div>
  <button
    class="emotion-0 emotion-1"
    type="button"
  >
    Test
  </button>
</div>
`;

exports[`Matches the transparent snapshot 1`] = `
.emotion-0 {
  -webkit-transition: filter 0.3s ease;
  transition: filter 0.3s ease;
  border: none;
  border-radius: var(--input-border-radius);
  font-size: inherit;
  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.25);
  margin: 5px 0px;
  color: inherit;
  background-color: transparent;
  box-shadow: none;
  padding: 0;
}

.emotion-0:hover,
.emotion-0:focus {
  cursor: pointer;
}

.emotion-0:active {
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
}

.emotion-0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(0.5);
  filter: grayscale(0.5);
}

<div>
  <button
    class="emotion-0 emotion-1"
    type="button"
  >
    Test
  </button>
</div>
`;
